nixos/modules/services/monitoring/grafana.nix

```diff
@@ -296,76 +204,85 @@ let
       secure_settings = mkOption {
         type = types.nullOr types.attrs;
         default = null;
-        description = lib.mdDoc "Secure settings for the notifier type.";
+        description = lib.mdDoc ''
+          Secure settings for the notifier type. Please note that the contents of this option
+          will end up in a world-readable Nix store. Use the file provider
+          pointing at a reasonably secured file in the local filesystem
+          to work around that. Look at the documentation for details:
+          <https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#file-provider>
+        '';
       };
     };
   };
 in {
+  imports = [
+    (mkRenamedOptionModule [ "services" "grafana" "protocol" ] [ "services" "grafana" "settings" "server" "protocol" ])
+    (mkRenamedOptionModule [ "services" "grafana" "addr" ] [ "services" "grafana" "settings" "server" "http_addr" ])
+    (mkRenamedOptionModule [ "services" "grafana" "port" ] [ "services" "grafana" "settings" "server" "http_port" ])
```

Comment:
I've just noticed that this warning also triggers for me in a config which doesn't seem to reference services.grafana.port itself anywhere.

The warning also shows up in `nix-build nixos/tests/grafana/basic.nix` among with a few others:

```
$ nix-build grafana/basic.nix
trace: Obsolete option `services.grafana.port' is used. It was renamed to `services.grafana.settings.server.http_port'.
trace: warning: The option `services.grafana.analytics.reporting.enable' defined in `makeTest parameters' has been renamed to `services.grafana.settings.analytics.reporting_enabled'.
trace: warning: The option `services.grafana.security.adminPassword' defined in `makeTest parameters' has been renamed to `services.grafana.settings.security.admin_password'.
trace: warning: The option `services.grafana.security.adminUser' defined in `makeTest parameters' has been renamed to `services.grafana.settings.security.admin_user'.
trace: warning: The option `services.grafana.domain' defined in `makeTest parameters' has been renamed to `services.grafana.settings.server.domain'.
trace: warning: The option `services.grafana.addr' defined in `makeTest parameters' has been renamed to `services.grafana.settings.server.http_addr'.
trace: warning: Grafana passwords will be stored as plaintext in the Nix store! Use file provider instead.
trace: warning: Provisioning Grafana datasources with options has been deprecated.
Use `services.grafana.provision.datasources.settings` or
`services.grafana.provision.datasources.path` instead.

trace: warning: Provisioning Grafana dashboards with options has been deprecated.
Use `services.grafana.provision.dashboards.settings` or
`services.grafana.provision.dashboards.path` instead.

trace: Obsolete option `services.grafana.port' is used. It was renamed to `services.grafana.settings.server.http_port'.
```

Reply:
Must be this:
```
      serviceConfig = {
        WorkingDirectory = cfg.dataDir;
        User = "grafana";
        RuntimeDirectory = "grafana";
        RuntimeDirectoryMode = "0755";
        # Hardening
        AmbientCapabilities = lib.mkIf (cfg.port < 1024) [ "CAP_NET_BIND_SERVICE" ];
        CapabilityBoundingSet = if (cfg.port < 1024) then [ "CAP_NET_BIND_SERVICE" ] else [ "" ];
        DeviceAllow = [ "" ];
        LockPersonality = true;
        NoNewPrivileges = true;
        PrivateDevices = true;
        PrivateTmp = true;
        ProtectClock = true;
        ProtectControlGroups = true;
        ProtectHome = true;
        ProtectHostname = true;
        ProtectKernelLogs = true;
        ProtectKernelModules = true;
        ProtectKernelTunables = true;
        ProtectProc = "invisible";
        ProtectSystem = "full";
        RemoveIPC = true;
        RestrictAddressFamilies = [ "AF_INET" "AF_INET6" "AF_UNIX" ];
        RestrictNamespaces = true;
        RestrictRealtime = true;
        RestrictSUIDSGID = true;
        SystemCallArchitectures = "native";
        # Upstream grafana is not setting SystemCallFilter for compatibility
        # reasons, see https://github.com/grafana/grafana/pull/40176
        SystemCallFilter = [ "@system-service" "~@privileged" ];
        UMask = "0027";
      };
```
It does reference `services.grafana.port`. I completely forgot about that.

Reply:
Feel free to cc me when you have some time to make the followup PR to this change and I'm happy to give it a run on my configs to make sure it's warning-clean! :)

Reply:
Dang, I also missed that, sorry! (perhaps partly because it's not bold red).